examples/petsc/qfunctions/bp1.h

```diff
@@ -34,14 +35,32 @@ CEED_QFUNCTION(SetupMass)(void *ctx, const CeedInt Q,
                             J[i+Q*1]*(J[i+Q*3]*J[i+Q*8] - J[i+Q*5]*J[i+Q*6]) +
                             J[i+Q*2]*(J[i+Q*3]*J[i+Q*7] - J[i+Q*4]*J[i+Q*6]));
     qdata[i] = det * w[i];
```

Comment:
@Jeremy Why the `qdata` output  here could not be reused as an input in the `SetupMassRhs` Q-function (avoiding to recalculate the Jacobian), as we do in the `Mass` Q-function?

Reply:
We definitely can; I just missed that.

Reply:
Actually, I did have a good reason for this - BP3-6 don't calculate the mass matrix qdata so I would have to carry extra qdata information that I will only use once or calculate the BP1-2 data an extra time.

Reply:
But even in `bp3.h` you are recalculating the Jacobian yet another time in the `SetupDiffRhs` Q-function, and similarly in `SetupDiffRhs3` in `bp4.h`.
 I would have left the qdata for BP3-6 unpacked (as we did for the navier-stokes example) and reused `w*detJ` that every Q-function uses